cmd/kubemark/hollow-node.go

```diff
@@ -181,52 +185,84 @@ func run(config *hollowNodeConfig) {
 	}
 	if config.ResourceServer == "" {
 		klog.V(3).Infof("Resource is not set. Default to clientConfig setting")
-		config.ResourceServer = clientConfigs.GetConfig().Host
+		config.ResourceServer = config.KubeconfigPath
 	}
 
 	// initialize the kubeclient manager
 	kubeclientmanager.NewKubeClientManager()
 
+	// TODO: set this from the hollow node env variable
+	enableInSecurePorts := false
+
 	numberTenantPartitions := len(config.TenantServers)
-	clients := make([]clientset.Interface, numberTenantPartitions)
-	for i := 0; i < len(clients); i++ {
-		tenantCfg := restclient.CopyConfigs(clientConfigs)
-		for _, cfg := range tenantCfg.GetAllConfigs() {
-			cfg.Host = config.TenantServers[i]
-		}
-		clientFromConfig, err := clientset.NewForConfig(tenantCfg)
-		if err != nil {
-			klog.Fatalf("Failed to create a ClientSet: %v. Exiting.", err)
+	client := make([]clientset.Interface, numberTenantPartitions)
+	for i := 0; i < len(client); i++ {
+		if enableInSecurePorts == false {
+			kubeconfigFile := config.TenantServers[i]
+			klog.V(2).Infof("create client config from file: %s", kubeconfigFile)
+			config, err := config.createClientConfigFromSpecificFile(kubeconfigFile)
+			if err != nil {
+				klog.Fatalf("Failed to create a client config: %v. Exiting.", err)
+			}
+
+			clientFromConfig, err := clientset.NewForConfig(config)
+			if err != nil {
+				klog.Fatalf("Failed to create a ClientSet: %v. Exiting.", err)
+			}
+			client[i] = clientFromConfig
+		} else {
+			tenantCfg := restclient.CopyConfigs(clientConfigs)
+			for _, cfg := range tenantCfg.GetAllConfigs() {
+				cfg.Host = config.TenantServers[i]
+			}
+			clientFromConfig, err := clientset.NewForConfig(tenantCfg)
+			if err != nil {
+				klog.Fatalf("Failed to create a ClientSet: %v. Exiting.", err)
+			}
+			client[i] = clientFromConfig
 		}
-		clients[i] = clientFromConfig
 	}
 
 	if config.Morph == "kubelet" {
 		f, c := kubemark.GetHollowKubeletConfig(config.createHollowKubeletOptions())
 
-		heartbeatClientConfigs := restclient.CopyConfigs(clientConfigs)
-		for _, heartbeatClientConfig := range heartbeatClientConfigs.GetAllConfigs() {
-			heartbeatClientConfig.Timeout = c.NodeStatusUpdateFrequency.Duration
-			// if the NodeLease feature is enabled, the timeout is the minimum of the lease duration and status update frequency
-			if utilfeature.DefaultFeatureGate.Enabled(features.NodeLease) {
-				leaseTimeout := time.Duration(c.NodeLeaseDurationSeconds) * time.Second
-				if heartbeatClientConfig.Timeout > leaseTimeout {
-					heartbeatClientConfig.Timeout = leaseTimeout
+		var heartbeatClient *clientset.Clientset
+		if enableInSecurePorts == false {
+			klog.V(2).Infof("create client config from file: %s", config.ResourceServer)
+			config, err := config.createClientConfigFromSpecificFile(config.ResourceServer)
+			if err != nil {
+				klog.Fatalf("Failed to create a client config: %v. Exiting.", err)
+			}
+
+			heartbeatClient, err = clientset.NewForConfig(config)
```

Comment:
I don't think client timeout and QPS are set in kubeconfig. Also need to check utilfeature.DefaultFeatureGate.Enabled(features.NodeLease) in secure mode

Reply:
Those are existing code and settings.